src/S3/PostObjectV4.php

```diff
@@ -151,8 +151,10 @@ private function generateUri()
             // Use path-style URLs
             $uri = $uri->withPath($this->bucket);
         } else {
-            // Use virtual-style URLs
-            $uri = $uri->withHost($this->bucket . '.' . $uri->getHost());
+            // Use virtual-style URLs if haven't been set up already
+            if (strpos($uri->getHost(), $this->bucket) === false) {
```

Comment:
Shouldn't this check only for prefix? Because what if they used a field with the same bucket name?